cl/recap/tasks.py

```diff
@@ -1838,6 +1844,99 @@ class DocketUpdatedData:
+    """
+
+    message_id = epq.message_id
+    bucket = RecapEmailSESStorage()
+    # Try to read the file using utf-8.
+    # If it fails fallback on iso-8859-1
+    try:
+        with bucket.open(message_id, "rb") as f:
+            body = f.read().decode("utf-8")
+    except UnicodeDecodeError:
+        with bucket.open(message_id, "rb") as f:
+            body = f.read().decode("iso-8859-1")
+    report = S3NotificationEmail(map_cl_to_pacer_id(epq.court_id))
+    report._parse_text(body)
+    data = report.data
+    if (
+        data == {}
+        or len(data["dockets"]) == 0
+        or len(data["dockets"][0]["docket_entries"]) == 0
+        or data["dockets"][0]["docket_entries"][0]["pacer_doc_id"] is None
+    ):
+        msg = "Not a valid notification email. No message content."
+        mark_pq_status(
+            epq, msg, PROCESSING_STATUS.INVALID_CONTENT, "status_message"
+        )
+        data = None
+    return data, body
+
+
+def get_and_merge_rd_attachments(
+    document_url: str,
+    court_id: str,
+    dockets_updated: list[DocketUpdatedData],
+    user_pk: int,
+) -> list[RECAPDocument]:
+    """Get the attachment page and merge the data into the dockets returned
+    by the recap.email notification.
+
+    :param document_url: The document URL including the magic number to get the
+     attachment page without being logged into PACER.
+    :param court_id: The court ID we're working with.
+    :param dockets_updated: A list of DocketUpdatedData containing the dockets
+    to merge the attachments in.
+    :param user_pk: The user to associate with the ProcessingQueue object.
+    :return: A list of RECAPDocuments modified or created during the process
+    """
+
+    all_attachment_rds = []
+    cookies = get_pacer_cookie_from_cache(user_pk)
+    # Try to get the attachment page without being logged into PACER
+    att_report_text = get_attachment_page_by_url(document_url, court_id)
+    if att_report_text:
+        att_report = AttachmentPage(court_id)
+    else:
+        main_rd = (
+            dockets_updated[0]
+            .des_returned[0]
+            .recap_documents.earliest("date_created")
+        )
+        # Get the attachment page being logged into PACER
+        att_report = get_attachment_page_by_rd(main_rd.pk, cookies)
+
+    for docket_entry in dockets_updated:
+        # Merge the attachments for each docket/recap document
+        main_rd_local = docket_entry.des_returned[0].recap_documents.earliest(
+            "date_created"
+        )
+        pq_pk = make_attachment_pq_object(
+            att_report,
+            main_rd_local.pk,
+            user_pk,
+            att_report_text,
+        )
+        # We only query and parse the Attachment page for one document in PACER
+        # The only field that needs to be overwritten is the document number
+        # since it changes from case to case. Assigns it from its main document
```

Comment:
Can you clarify that this to explain how it applies only to multi-docket notifications?